### ISSUE TITLE
Domains: "Attach to an existing site" and transfer domain to another site options not working

### ISSUE THREAD

**Issue by jartes** [Automattic/wp-calypso#90508]
### Quick summary

When trying to attach a registered domain to any site or when trying to transfer a domain to another WordPress.com site, the screen goes blank.

### Steps to reproduce

For the "Attach to an existing site" issue:
1. Go to https://wordpress.com/domains/manage
2. Search for a domain with no site attached
3. On the three-dots button option, click on the `Attach to an existing site`. 

For the domain transfer between sites: 
1. Go to https://wordpress.com/domains/manage
2. Click on a domain with an attached site. 
3. Click the `Transfer` option
4. Select the option "To another WordPress.com site"

### What you expected to happen

In both scenarios, get the proper screen to attach or transfer the domain to the corresponding site.

### What actually happened

Got a blank screen, with the following error: 

```
TypeError: Cannot read properties of undefined (reading 'manage_options')
    at domains.ab7e5994ffd67b739880.min.js:1:378685
    at Array.filter (<anonymous>)
    at ei.sitesToBeRendered (24873.03c9d408d88b25905753.min.js:1:29992)
    at ei.render (24873.03c9d408d88b25905753.min.js:1:31697)
    at im (18543.590520b334250fcd9f1a.min.js:25:75793)
    at ih (18543.590520b334250fcd9f1a.min.js:25:75591)
    at l (18543.590520b334250fcd9f1a.min.js:25:120382)
    at aL (18543.590520b334250fcd9f1a.min.js:25:99057)
    at 18543.590520b334250fcd9f1a.min.js:25:98924
    at az (18543.590520b334250fcd9f1a.min.js:25:98931)
```

### Impact

All

### Available workarounds?

Yes, difficult to implement

### Platform (Simple and/or Atomic)

Simple, Atomic

### Logs or notes

This was escalated internally here p1715263135548939-slack-C02FMH4G8

**Comment by jartes** [comment]
In this chat, user reported this when trying to attach a domain to one existing simple site: 8168365-zd-a8c

**Comment by jartes** [comment]
8168166-zd-a8c

**Comment by delputnam** [comment]
Fixed in #90507
